main.go

```diff
@@ -151,6 +155,46 @@ func updateNDK(revision string) error {
 	return nil
 }
 
+func unWrapNDKDirIfNeeded(ndkHome string, revision string) error {
+	platformDirName := "platforms"
+	platformDir := filepath.Join(ndkHome, platformDirName)
+	p, err := pathutil.IsDirExists(platformDir)
+	if err != nil {
+		return fmt.Errorf("could not check if directory %s exists. Error: %v", platformDir, err)
+	}
+	if p {
+		return nil
+	}
+	revDirName := fmt.Sprintf("android-ndk-r%s", revision)
+	revDir := filepath.Join(ndkHome, revDirName)
+	r, err := pathutil.IsDirExists(revDir)
+	if err != nil {
+		return fmt.Errorf("could not check if directory %s exists. Error: %v", revDir, err)
+	}
+	if !r {
+		return fmt.Errorf("directory %s not exists", revDir)
+	}
+	revPlatformDir := filepath.Join(revDir, platformDirName)
+	rp, err := pathutil.IsDirExists(revPlatformDir)
+	if err != nil {
+		return fmt.Errorf("could not check if directory %s exists. Error: %v", revDir, err)
+	}
+	if rp {
+		return unWrapNDKDir(revDir, ndkHome)
+	}
+	return fmt.Errorf("could not locate platform dir at ndk home: %s", ndkHome)
+}
+
+func unWrapNDKDir(src string, dst string) error {
```

Comment:
The function name unWrapNDKDir is not an exact description of the. The logic in the function is copying a dir recursively and removes the old dir. So the name could be something like "move". However this could happen without copying and removing (which is kinda resource heavy task), you can use https://golang.org/pkg/os/#Rename to change the path of a given directory. (In UNIX the directory and file path is the "Exact" name of the pointer for the data source on the disk. This is why you can rename the path itself and not only the file's name.)

Reply:
Changed to move, refactored named accordingly.